### PR TITLE
Fix translations unavailable in defaultState rendering

### PR DESCRIPTION
**Pull request by sandlerr**
- switch to default state after loading translations
- remove use of app in I18n module
- document I18n module

@zendesk/vegemite @joseconsador

### CODE REVIEW

File: lib/javascripts/base_app.js [hunk 1]
```diff
@@ -46,17 +46,17 @@ function registerHelpers(app) {
...
   let evt = { firstLoad: true };
   this._metadata = data.metadata;
   this._context = data.context;
-  if (this.defaultState) {
-    this.switchTo(this.defaultState);
-  }
   zafClient.get('currentUser.locale').then(function(data) {
-    I18n.loadTranslations(this, data['currentUser.locale']);
+    I18n.loadTranslations(data['currentUser.locale']);
+    if (this.defaultState) {
```
Comment (sandlerr) [inline lib/javascripts/base_app.js]: @joseconsador this is the actual fix - move the `switchTo(this.defaultState)` after `loadTranslations`

File: doc/i18n.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+### Initialization
+
+When you know which locale you want to use, call `I18n.loadTranslations(locale)` where
+`locale` is a string like `en-US`, `de`, `zh`. This will load the strings under the
+matching file in `src/translations`. For example, with a configured `zafClient` object
+you could use
+
+```javascript
+zafClient.get('currentUser.locale').then(function(data) {
```
Comment (danielbreves) [inline doc/i18n.md]: Maybe add the init call above this?
```js
var client = ZafClient.init();
```

Comment (sandlerr) [inline doc/i18n.md]: done

File: doc/i18n.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  I18n.loadTranslations(data['currentUser.locale']);
+});
+```
+
+### Usage
```
Comment (danielbreves) [inline doc/i18n.md]: Can we write this with the reference format? e.g.

```md
### I18n.t(key, context)

What this does.

#### Arguments

  * `something` the thing you pass

#### Returns

The thing it returns.
```

**Comment by danielbreves** [comment]
👍

File: doc/i18n.md (hunk 3 above)
Comment (sandlerr) [inline doc/i18n.md]: ok, same format for `loadTranslations` too?

**Comment by joseconsador** [comment]
👍

**Comment by sandlerr** [comment]
@danielbreves 💇

File: doc/i18n.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+Sets the locale to be used by `I18n.t()`
+
+#### Arguments
+
+* `locale`: String representing the filename of the required translation JSON file.
+
+#### Returns
+
+`undefined`
```
Comment (danielbreves) [inline doc/i18n.md]: I wouldn't include the `Returns` header if it doesn't return anything meaningful.

Comment (sandlerr) [inline doc/i18n.md]: removed

**Comment by danielbreves** [comment]
cc @skipjac

File: doc/i18n.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
+## Using the I18n module
+
+**This document does not apply to apps that are migrated from v1 of the Zendesk
+App Framework.**
+
+The I18n module provides a helper called `I18n.t` that you can use to localize
+your application.
```
Comment (adammw) [inline doc/i18n.md]: Should we mention the intention of the module is to assist migration, and people should consider using a premade i18n library for newly developed apps?

**Pull request merged**